internal/packages/internal/packagekickstart/kickstart.go

```diff
@@ -0,0 +1,125 @@
+package packagekickstart
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"pkg.package-operator.run/cardboard/kubeutils/kubemanifests"
+	"sigs.k8s.io/yaml"
+
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+	manifestsv1alpha1 "package-operator.run/apis/manifests/v1alpha1"
+	"package-operator.run/internal/packages/internal/packagetypes"
+)
+
+type KickstartResult struct {
+	ObjectCount             int
+	GroupKindsWithoutProbes []schema.GroupKind
+}
+
+func Kickstart(_ context.Context, pkgName string, objects []unstructured.Unstructured) (
+	*packagetypes.RawPackage, KickstartResult, error,
+) {
+	res := KickstartResult{}
+	rawPkg := &packagetypes.RawPackage{
+		Files: packagetypes.Files{},
+	}
+
+	// Process objects
+	usedPhases := map[string]struct{}{}
+	usedGKs := map[schema.GroupKind]struct{}{}
+	var objCount int
+	for _, obj := range objects {
+		annotations := obj.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		gk := obj.GroupVersionKind().GroupKind()
+		phase := guessPresetPhase(gk)
+		annotations[manifestsv1alpha1.PackagePhaseAnnotation] = phase
+		obj.SetAnnotations(annotations)
+
+		path := filepath.Join(phase,
+			fmt.Sprintf("%s.%s.yaml", obj.GetName(),
+				strings.ToLower(obj.GetKind())))
+		b, err := yaml.Marshal(obj.Object)
+		if err != nil {
+			return nil, res, fmt.Errorf("marshalling YAML: %w", err)
+		}
+		rawPkg.Files[path] = b
+
+		usedPhases[phase] = struct{}{}
+		usedGKs[gk] = struct{}{}
+		objCount++
+	}
+
+	// Generate Manifest
+	var phases []manifestsv1alpha1.PackageManifestPhase
+	for _, phase := range orderedPhases {
+		if _, ok := usedPhases[string(phase)]; ok {
+			phases = append(phases, manifestsv1alpha1.PackageManifestPhase{Name: string(phase)})
+		}
+	}
+	//nolint:prealloc
+	var (
+		probes           []corev1alpha1.ObjectSetProbe
+		gksWithoutProbes = map[schema.GroupKind]struct{}{}
+	)
```

Comment:
:bike: :shed: ignore if you don't like it

```suggestion
	probes := []corev1alpha1.ObjectSetProbe{}
	gksWithoutProbes := map[schema.GroupKind]struct{}{}
```